src/vs/editor/contrib/codeAction/browser/codeActionMenu.ts

```diff
@@ -71,6 +71,7 @@ export interface ICodeActionMenuItem {
 	isEnabled: boolean;
 	isDocumentation: boolean;
 	index: number;
+	disabledReason?: string;
```

Comment:
Instead of passing through more properties of the action, how about just passing through the code action itself?

Reply:
I added it in the `ICodeActionMenuItem` since I have to check in `renderElement` of the list widget.

``` typescript
if (element.disabledReason)
	data.root.title = element.disabledReason;
```

One alternative though, if I passed in the entire code action (as an IAction or CodeActionAction), I could make the checks and the parsing of the code action all in the renderElement function. I think it might make it a bit cluttered though with the type casting and everything in the renderElement function.

Reply:
Yes give that a try. It would also let you remove properties like `isDocumentation` and instead compute their value in render function. Will requires some refactoring but should be clearer long term